types/instance.d.ts

```diff
@@ -82,12 +82,117 @@ export interface FastifyInstance<
   inject(opts: InjectOptions | string): Promise<LightMyRequestResponse>;
   inject(): LightMyRequestChain;
 
+  /**
+   * @deprecated Variadic listen method is deprecated. Please use `.listen(optionsObject)` instead. The variadic signature will be removed in `fastify@5`
+   * @see https://github.com/fastify/fastify/pull/3712
```

Comment:
Suggestions are welcome.
For now I choose to display the same message as the warning and to link the GitHub PR acting this change.